Darwin/StitchCore/StitchCore/Core/Auth/Internal/StitchAuthImpl.swift

```diff
@@ -303,44 +301,35 @@ internal final class StitchAuthImpl: CoreStitchAuth<StitchUserImpl>, StitchAuth
     // MARK: Observer Delegates
 
     /**
-     * Registers a `StitchAuthDelegate` with the client. The `StitchAuthDelegate`'s `onAuthEvent(:fromAuth)`
-     * method will be called with this `StitchAuth` as the argument whenever this client is authenticated
-     * or is logged out.
-     *
-     * - important: StitchAuthDelegates registered here are stored as `weak` references, meaning that if there are no
-     *              more strong references to a provided delegate, its `onAuthEvent(:fromAuth)` method will no longer
-     *              be called on authentication events.
-     * - parameters:
-     *     - authDelegate: A class conforming to `StitchAuthDelegate`, whose `onAuthEvent(:fromAuth)` method should be
-     *                     called whenever this client experiences an authentication event.
-     */
-    public func add(authDelegate: StitchAuthDelegate) {
-        objc_sync_enter(self)
-        self.delegates.append(DelegateWeakRef(value: authDelegate))
-        objc_sync_exit(self)
-
-        // Trigger the onUserLoggedIn event in case some event happens and
-        // this caller would miss out on this event other wise.
-        dispatcher.queue.async {
-            authDelegate.onAuthEvent(fromAuth: self)
-        }
-    }
-
-    /**
-     * Calls the `onAuthEvent` method of each registered `StitchAuthDelegate`.
+     * Dispatches the appropriate auth event method of each registered `StitchAuthDelegate`.
      *
      * - important: This is not meant to be invoked directly in this class. The `CoreStitchAuth` from which this
      *              class inherits will call this method when appropraite.
      */
-    public final override func onAuthEvent() {
-        self.delegates.enumerated().reversed().forEach { idx, delegateRef in
-            guard let delegate = delegateRef.value else {
+    public final override func dispatchAuthEvent(_ authEvent: AuthRebindEvent) {
+        // NOTE: this function is in this file rather than the +Delegation file because Swift does not support
+        //       overriding functions in extensions
```

Comment:
How come we iterate through backwords?

Reply:
this way, when we delete something while we're iterating through the list of delegates, we still see every item and don't accidentally miss one because the index wasn't updated when we deleted something

see: https://stackoverflow.com/questions/1582285/how-to-remove-elements-from-a-generic-list-while-iterating-over-it

Reply:
interesting, thanks